Call to virtual function during destruction in node_thread.cc

Warning from `clang-tidy`:

```
oak/server/node_thread.cc:23:29: warning: Call to virtual function during destruction [clang-analyzer-optin.cplusplus.VirtualCall]
NodeThread::~NodeThread() { Stop(); }
                            ^
oak/server/node_thread.cc:23:29: note: This destructor of an object of type '~NodeThread' has not returned when the virtual method was called
oak/server/node_thread.cc:23:29: note: Call to virtual function during destruction
```

--- a/oak/server/node_thread.cc
+++ b/oak/server/node_thread.cc
@@ -20,7 +20,7 @@
 
 namespace oak {
 
-NodeThread::~NodeThread() { Stop(); }
+NodeThread::~NodeThread() { StopThread(); }
 
 void NodeThread::Start(Handle handle) {
   if (thread_.joinable()) {
@@ -37,7 +37,9 @@ void NodeThread::Start(Handle handle) {
   LOG(INFO) << "Started {" << name_ << "} node thread";
 }
 
-void NodeThread::Stop() {
+void NodeThread::Stop() { StopThread(); }
+
+void NodeThread::StopThread() {
   termination_pending_ = true;
   if (thread_.joinable()) {
     LOG(INFO) << "Waiting for completion of {" << name_ << "} node thread";

--- a/oak/server/node_thread.h
+++ b/oak/server/node_thread.h
@@ -44,6 +44,8 @@ class NodeThread : public OakNode {
   virtual void Run(Handle handle) = 0;
 
  private:
+  void StopThread();
+
   std::thread thread_;
 };
 

